src/proxy/http/HttpSM.cc

```diff
@@ -5193,9 +5193,8 @@ HttpSM::get_outbound_sni() const
 
   if (policy.empty() || policy == "host"_tv) {
     // By default the host header field value is used for the SNI.
-    int         len;
-    char const *ptr = t_state.hdr_info.server_request.host_get(&len);
-    zret.assign(ptr, len);
+    auto host{t_state.hdr_info.server_request.host_get()};
+    zret.assign(host.data(), static_cast<int>(host.length()));
```

Comment:
I think this could be: `zret = t_state.hdr_info.server_request.host_get()`

Reply:
Thanks! Fixed in [9e31e96](https://github.com/apache/trafficserver/pull/12091/commits/9e31e96013a96779e3963a2dff9cf764da017554)